articles/connecting-library-application-to-sql-server-using-windows-forms/index.md

```diff
@@ -0,0 +1,232 @@
+          foreach (DataRow dr in dt.Rows)
+        { dataGridView1.Rows.Add(dr["title_id"], dr["pub_name"], dr["price"], dr["ytd_sales"], dr["title"], dr["type"]);}}
+```
+
+### Authors tab
+
+The author's tab will display the id first name phone address city & count of each book the author wrote.
+
+![Authors](/engineering-education/connecting-library-application-to-sql-server-using-windows-forms/Authors.png).
+
+```c#
+ string sqlQuery2 = "select * from authors order by au_fname asc";
+            DataTable dt2 = DataBaseConnection.dataAdapterSelect(sqlQuery2);
+            foreach (DataRow dr in dt2.Rows)
+            {dataGridView2.Rows.Add(dr["au_id"], dr["au_fname"], dr["phone"], dr["address"], dr["city"]);}
+            
+              string sqlQuery3 = "SELECT authors.au_fname,count(titleauthor.title_id) as authorcount FROM authors, titleauthor WHERE authors.au_id = titleauthor.au_id GROUP BY authors.au_fname";
+            DataTable dt3 = DataBaseConnection.dataAdapterSelect(sqlQuery3);
+            foreach (DataRow dr in dt3.Rows)
+            {dataGridView3.Rows.Add(dr["authorcount"]);}
+```
+### Publishers tab
+
+The Publisher's tab will display the information of the publisher using the id.
+
+![Publishers](/engineering-education/connecting-library-application-to-sql-server-using-windows-forms/Publishers.png).
+
+```c#
+private void checkpub_Click(object sender, EventArgs e)
+            { string idd = puid.Text; 
+            string sqlQuery00 = "select * from  publishers where publishers.pub_id='" + idd + "'";
+            DataTable dt00 = DataBaseConnection.dataAdapterSelect(sqlQuery00);
+
+            if (dt00.Rows.Count > 0)
+            {foreach (DataRow dr in dt00.Rows)
+                { dataGridView4.Rows.Clear();
+                  dataGridView4.Rows.Add(dr["pub_id"], dr["pub_name"], dr["city"], dr["state"], dr["country"]);}}
+            else
+            {MessageBox.Show("This Id is does not exist!");}}
+```
+
+### Store tab
+
+The Store's tab will display the discounts available using a left outer join Query.
+
+![Store](/engineering-education/connecting-library-application-to-sql-server-using-windows-forms/Store.png).
+
+```c#
+string q5 = " select * from stores left outer join discounts on stores.stor_id = discounts.stor_id ";           
+            DataTable q55 = DataBaseConnection.dataAdapterSelect(q5);
+            
+            foreach (DataRow dr in q55.Rows)
+            {
+                dataGridView5.Rows.Add(dr["stor_name"], dr["discount"]);
+            }
+```
+
+### Conclusion
+
+In this tutorial we have learned how to connect an application to a database and convert any string to a Query that can be understood by the SQL server, we have also learned how to add edit & delete a book from the database & display the information of the books authors publishers & the discounts of each store. Don't forget to test out the code to fully understand how it works.

```

Comment:
Add a `Further reading` section and reference few links which would help the reader gain more knowledge about this topic